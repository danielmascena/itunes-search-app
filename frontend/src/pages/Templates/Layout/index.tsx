import * as React from "react";
import Box from "@mui/material/Box";

import Header from "../Header";
import Footer from "../Footer";
import ErrorBoundary from "../../../components/ErrorBoundary";

interface Props {
  children: React.ReactNode;
}

const Layout = ({ children }: Props) => (
  <ErrorBoundary>
    <Header />
    <Box component="main">{children}</Box>
    <Footer />
  </ErrorBoundary>
);

export default Layout;
