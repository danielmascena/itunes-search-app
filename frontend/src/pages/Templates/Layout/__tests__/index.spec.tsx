import * as React from "react";
import renderer from "react-test-renderer";
import Box from "@mui/material/Box";

import Layout from "../index";

test("renders the Layout component", () => {
  expect(
    renderer
      .create(
        <Layout>
          <Box>Basic Layout</Box>
        </Layout>
      )
      .toJSON()
  ).toMatchSnapshot();
});
