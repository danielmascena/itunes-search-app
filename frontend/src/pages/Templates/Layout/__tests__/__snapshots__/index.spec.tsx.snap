// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Layout component 1`] = `
Array [
  <header
    className="MuiBox-root css-0"
  >
    <h5
      className="MuiBox-root css-1yuhvjn"
    >
      iTunes Search API
    </h5>
  </header>,
  <main
    className="MuiBox-root css-0"
  >
    <div
      className="MuiBox-root css-0"
    >
      Basic Layout
    </div>
  </main>,
  <footer
    className="MuiBox-root css-0"
  >
    <p
      className="MuiBox-root css-0"
    >
      Copyright Â©
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-zr39bh-MuiTypography-root-MuiLink-root"
        href="https://github.com/danielmascena"
        onBlur={[Function]}
        onFocus={[Function]}
        title="author"
      >
        Daniel Mascena
      </a>
    </p>
  </footer>,
]
`;
